### PR TITLE
feat(azure): attach nics to running instance (SC-1712)

### PR DESCRIPTION
**Pull request by CalvoM**
Test script
```python
import logging
import time
from pycloudlib import Azure, azure


def main_multi_nic():
    rg = azure.util.AzureParams("main_sa_north", None)
    cloud = Azure(
        tag="calvinmwadime", region="southcentralus", resource_group_params=rg
    )
    inst = cloud.launch(cloud.daily_image("jammy"), name="JammyMultiDos")
    print(f"Launched {inst.name} {inst.status}")
    print(f"NICs: {inst.nics}")
    inst.deallocate()
    # REPLACE WITH NEW NIC TO ATTACH
    new_nic = {
        "id": "/subscriptions/1b7499f2-637c-46b5-b448-c97e9267e522/resourceGroups/main_sa_north/providers/Microsoft.Network/networkInterfaces/jammy_dos",
        "primary": False,
    }
    inst.attach_nics([new_nic])
    inst.start()


if __name__ == "__main__":
    logging.basicConfig(level=logging.DEBUG, filename="out1.boot_logs")
    main_multi_nic()
```

### CODE REVIEW

**Comment by AleksaSvitlica** [comment]
Should this MP implement `add_network_interface` and `remove_network_interface` since they are defined in BaseInstance? That seems like it might better align with the rest of pycloudlib.

**Comment by CalvoM** [comment]
Oh, I had totally overlooked that. Let me dive deeper into that and make necessary changes.

File: VERSION [hunk 1]
```diff
@@ -1 +1 @@
-1!6.0.3
+1!6.0.4
```
Comment (aciba90) [inline VERSION:1]: Bump minor as we are introducing new features.

**Review by aciba90 — CHANGES_REQUESTED** [review]
Thanks, @CalvoM!

> Should this MP implement add_network_interface and remove_network_interface since they are defined in BaseInstance? That seems like it might better align with the rest of pycloudlib.

As @AleksaSvitlica said, I think we should adhere to / extend the  `add_network_interface` and `remove_network_interface` interfaces of `BaseInstance`.

File: pycloudlib/azure/instance.py [hunk 2]
```diff
@@ -164,3 +176,21 @@ def delete(self, wait=True) -> List[Exception]:
             return [e]
 
         return []
+
+    def attach_nics(self, nics: List[Dict[str, str]]):
+        """Attach nics to instance."""
+        if self.status != "stopped":
+            raise PycloudlibError(
+                "VM must be deallocated before attaching NIC"
```
[the remark this answers is not in the document]
Comment (CalvoM) [reply pycloudlib/azure/instance.py:184]: I have reviewed this and it does not need to be deallocated but just stopped.

[the remark this answers is not in the document]
Comment (CalvoM) [reply pycloudlib/azure/instance.py:197]: After testing, we need to deallocate to add/remove nics.

**Review by aciba90 — APPROVED** [review]
LGTM, thanks!

**Pull request merged**